bpo-41861: Clean up sqlite3 header files wrt. PEP 384

erlend-aasland opened the pull request:
Ref. https://www.python.org/dev/peps/pep-0384/#header-files-and-preprocessor-definitions
~Also ref. [bpo-2897](https://bugs.python.org/issue2897) (PyMemberDef not part of the stable API)~

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->


<!-- issue-number: [bpo-41861](https://bugs.python.org/issue41861) -->
https://bugs.python.org/issue41861
<!-- /issue-number -->

Commit:
Don't include pythread.h

Commit:
Don't include structmember.h twice

Commit:
Remove superfluous Python.h include

vstinner commented on Modules/_sqlite/connection.c:
```diff
@@ -23,7 +23,6 @@
 
 #include "cache.h"
 #include "module.h"
-#include "structmember.h"         // PyMemberDef
```
If PyMemberDef is only used in the C file, IMO it's better to not add it to the H file.

erlend-aasland replied:
Good point. I'll fix.

erlend-aasland replied:
Put it back to where it was in commit ~c9357792be~ 6be86c9882a33f27ac25d50880edf6035ab230b1

erlend-aasland replied:
Ah, I need to fix this again. PyMemberDef is used in two files... I'll add this include to `cursor.c` as well, so it's more explicit.

erlend-aasland replied:
… and `statement.c`

erlend-aasland replied:
Sorry, I had to rebase and force-push. This is all fixed in 6be86c9882a33f27ac25d50880edf6035ab230b1, f184612542abe9a1d03a4731a30059c7ac16e517, 926e2386edb3482e0879a9489ef1425d8692e9a0, and b98761c865a729849e13dbd134d0b4380ce0af94. PTAL and sorry 'bout the force push!

Commit:
Include structmember.h in the .c file, where it's used

As per review comments.

Commit:
Explicitly include structmember.h in cursor.c

Commit:
Include structmember.h after Python.h

Commit:
Fix PyMemberDef in statement.c

- Fix incomplete type (PyMemberDef => struct PyMemberDef)
- Include structmember.h bco. PyMemberDef
